test/test_datasets.py

```diff
@@ -2205,5 +2205,49 @@ def inject_fake_data(self, tmpdir: str, config):
         return len(sampled_classes * n_samples_per_class)
 
 
+class Country211TestCase(datasets_utils.ImageDatasetTestCase):
+    DATASET_CLASS = datasets.Country211
+    FEATURE_TYPES = (PIL.Image.Image, int)
+
+    ADDITIONAL_CONFIGS = datasets_utils.combinations_grid(split=("train", "valid", "test"))
+
+    def inject_fake_data(self, tmpdir: str, config):
+        root_folder = pathlib.Path(tmpdir) / "country211"
+        root_folder.mkdir()
+
+        num_images_per_class = 5
+        sampled_classes = ("AD", "BS", "GR")
+        im_paths = []
+
+        for cls in sampled_classes:
+            image_folder = root_folder / cls
+            im_paths.extend(
+                datasets_utils.create_image_folder(
+                    image_folder,
+                    name=cls,
+                    file_name_fn=lambda idx: f"{cls}_{idx:05d}.jpg",
+                    num_examples=num_images_per_class,
+                )
+            )
```

Comment:
1. As is, this will create images with the following path `{tmpdir}/country211/{cls}/{cls}`. So the class name occurs twice, while the split does not occur at all. 
2. The file name does not reflect the actual data. If you want it to be accurate it should be something like `f"{:7d}_{:.6f}_{:.6f}"` with all three randomly generated. On the other hand, the file name is irrelevant for the dataset. Thus, I suggest to go with the simplest choice and only append `.jpg` to the index.

```suggestion
        split_folder = pathlib.Path(tmpdir) / "country211" / config["split"]
        split_folder.mkdir(parents=True, exist_ok=True)

        return sum(
            [
                len(
                    datasets_utils.create_image_folder(
                        split_folder,
                        name=cls,
                        file_name_fn=lambda idx: f"{idx}.jpg",
                        num_examples=5,
                    )
                )
                for cls in ("AD", "BS", "GR")
            ]
        )
```